src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -0,0 +1,1101 @@
+        }
+    }
+
+    /// <inheritdoc/>
+    public void EndSession()
+    {
+        // Session means vstest.console process in the original api
+        // we don't have a process to manage, we are in-process.
+    }
+
+    /// <inheritdoc/>
+    public void StartSession()
+    {
+        // Session means vstest.console process in the original api
+        // we don't have a process to manage, we are in-process.
+    }
+
+    /// <inheritdoc/>
+    [Obsolete("This API is not final yet and is subject to changes.", false)]
+    public ITestSession? StartTestSession(
+        IList<string> sources,
+        string? runSettings,
+        ITestSessionEventsHandler eventsHandler)
+    {
+        return StartTestSession(sources, runSettings, options: null, eventsHandler);
+    }
+
+    /// <inheritdoc/>
+    [Obsolete("This API is not final yet and is subject to changes.", false)]
+    public ITestSession? StartTestSession(
+        IList<string> sources,
+        string? runSettings,
+        TestPlatformOptions? options,
+        ITestSessionEventsHandler eventsHandler)
+    {
+        return StartTestSession(
+            sources,
+            runSettings,
+            options,
+            eventsHandler,
+            testHostLauncher: null);
+    }
+
+    /// <inheritdoc/>
+    [Obsolete("This API is not final yet and is subject to changes.", false)]
+    public ITestSession? StartTestSession(
+        IList<string> sources,
+        string? runSettings,
+        TestPlatformOptions? options,
+        ITestSessionEventsHandler eventsHandler,
+        ITestHostLauncher? testHostLauncher)
+    {
+        _testPlatformEventSource.TranslationLayerStartTestSessionStart();
+
+        var resetEvent = new ManualResetEvent(false);
+        TestSessionInfo? testSessionInfo = null;
+
+        try
+        {
+            TestRequestManager?.ResetOptions();
+            var startTestSessionPayload = new StartTestSessionPayload()
+            {
+                Sources = sources,
+                RunSettings = runSettings,
+                HasCustomHostLauncher = testHostLauncher != null,
+                IsDebuggingEnabled = testHostLauncher != null
+                                     && testHostLauncher.IsDebug,
+                TestPlatformOptions = options
+            };
+
+            var inProcessEventsHandler = new InProcessTestSessionEventsHandler(eventsHandler);
+            inProcessEventsHandler.StartTestSessionCompleteEventHandler += (_, eventArgs) =>
+            {
+                testSessionInfo = eventArgs?.TestSessionInfo;
+                resetEvent.Set();
+            };
+
+            TestRequestManager?.StartTestSession(
+                startTestSessionPayload,
+                (ITestHostLauncher3?)testHostLauncher,
+                inProcessEventsHandler,
+                new ProtocolConfig { Version = _highestSupportedVersion });
+
+            resetEvent.WaitOne();
```

Comment:
some timeout?Usually we should add timeout/verify/throw or it's hard to understand in case of hang where we are.

Reply:
Fixed, thanks.